openhands-sdk/openhands/sdk/agent/agent.py

```diff
@@ -249,7 +244,10 @@ def step(
                 action_events.append(action_event)
 
             # Handle confirmation mode - exit early if actions need confirmation
-            if self._requires_user_confirmation(state, action_events):
+            if self._security_service.requires_confirmation(action_events):
+                state.execution_status = (
+                    ConversationExecutionStatus.WAITING_FOR_CONFIRMATION
+                )
```

Comment:
The original method handled both decision-making and state mutation, but now they're split across classes. This split could lead to inconsistencies if the agent forgets to set the execution status, or if the logic gets out of sync. Any ideas on how we may address this?

Reply:
Oh, the reason for this adjustment is to avoid modifying external resources. Instead, we leave the right to make modifications to the caller, preserving the read-only nature of this method. 
If needed, I think we can implement a method in DefaultSecurityService to handle the state.